tests/dsp-contracts.test.js

```
@@ -0,0 +1,162 @@
+    dispatched = [];
+    // Simulate no worklet node on window
+    delete global.window;
+    global.window = {};
+  });
+
+  test('does not throw when worklet is undefined', () => {
+    // We cannot import ES module in CJS without a bundler, so test the
+    // logic inline as a smoke-test contract.
+    const _pending = [];
+    function dispatchParam(id, rawVal, app) {
+      const resolved = app || window._vipApp || window.vip || window._vipOrch;
```

ESLint

[error] 'window' is not defined.

(no-undef)